lib/boavizta.ts

```diff
@@ -0,0 +1,189 @@
+import {IImpactModelInterface} from "./interfaces/index";
+import axios from "axios";
+
+export {
+    IImpactModelInterface
+} from "./interfaces/index";
+
+
+abstract class BoaviztaImpactModel implements IImpactModelInterface {
+    protected authCredentials: object | undefined;
+    name: string | undefined;
+    sharedParams: object | undefined = undefined;
+    metricType: "cpu" | "gpu" | "ram" = "cpu";
+
+
+    authenticate(authParams: object) {
+        this.authCredentials = authParams;
+    }
+
+    async configure(name: string, staticParams: object | undefined = undefined): Promise<IImpactModelInterface> {
+        this.name = name;
+        this.sharedParams = await this.captureStaticParams(staticParams ?? {});
+        return this
+    }
+
+    //abstract subs to make compatibility
+    protected abstract captureStaticParams(staticParams: object): any
+
+    abstract modelIdentifier(): string
+
+    abstract fetchData(usageData: object | undefined): Promise<object>
+
+
+    async supportedLocations(): Promise<string[]> {
```

Comment:
I don't think you're using this anywhere right?

Reply:
This function is for retrieval of data.